### ISSUE TITLE
Add provided.al2023 lambda runtime

### ISSUE THREAD

**Issue by edoardocostantinidev** [pulumi/pulumi-aws#2994]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

AWS introduced a [new](https://aws.amazon.com/blogs/compute/introducing-the-amazon-linux-2023-runtime-for-aws-lambda/) lambda runtime `provided.al2023`. If you try and set a lambda with that runtime you receive the following error:
```
expected runtime to be one of [nodejs nodejs4.3 nodejs6.10 nodejs8.10 nodejs10.x nodejs12.x nodejs14.x nodejs16.x java8 java8.al2 java11 python2.7 python3.6 python3.7 python3.8 python3.9 dotnetcore1.0 dotnetcore2.0 dotnetcore2.1 dotnetcore3.1 dotnet6 nodejs4.3-edge go1.x ruby2.5 ruby2.7 provided provided.al2 nodejs18.x python3.10 java17], got provided.al2023.
```

### Affected area/feature

missing provider option for lambda runtime.

**Comment by mikhailshilkov** [comment]
The upstream change has shipped two hours ago, we'll probably get it next week https://github.com/hashicorp/terraform-provider-aws/issues/34360

**Comment by jbvv1** [comment]
The required upstream change was released as part of v5.26.0 of the AWS Terraform provider on November 16, 2023: https://github.com/hashicorp/terraform-provider-aws/issues/34360#issuecomment-1815494950. When will Pulumi get updated to support the `provided.al2023` runtime?
